src/getIdentifier.ts

```diff
@@ -0,0 +1,92 @@
+import { Record, String, Static } from "runtypes";
+import { VerificationFragment } from "./types/core";
+import { Identity } from "./verifiers/issuerIdentity/dnsTxt";
+import { SignatureVerificationFragment } from "./verifiers/issuerIdentity/did";
+import { DnsVerificationFragment } from "./verifiers/issuerIdentity/dnsDid";
+
+enum IdentityProof {
+  DNS = "OpenAttestationDnsTxtIdentityProof",
+  DNSDID = "OpenAttestationDnsDidIdentityProof",
+  DID = "OpenAttestationDidIdentityProof",
+}
+
+const DataDnsIdentifier = Record({
+  identifier: String,
+  value: String,
+});
+type DataDnsIdentifier = Static<typeof DataDnsIdentifier>;
+
+const getDnsIdentifierProof = (fragment: VerificationFragment<DataDnsIdentifier | Identity[]>) => {
+  const type = "DNS";
+  if (Array.isArray(fragment.data)) {
+    return fragment.data.map((issuer: Identity) => ({
+      identifier: issuer.location,
+      type,
+    }));
+  }
+  return {
+    identifier: fragment.data?.identifier,
+    type,
+  };
+};
+
+const getDnsDidIdentifierProof = (
+  fragment: VerificationFragment<DnsVerificationFragment | DnsVerificationFragment[]>
+) => {
+  const type = "DNS-DID";
+  if (Array.isArray(fragment.data)) {
+    return fragment.data.map((issuer: DnsVerificationFragment) => ({
+      identifier: issuer.location,
+      type,
+    }));
+  }
+  return {
+    identifier: fragment.data?.location,
+    type,
+  };
+};
+
+const getDidIdentifierProof = (
+  fragment: VerificationFragment<SignatureVerificationFragment | SignatureVerificationFragment[]>
+) => {
+  const type = "DID";
+  if (Array.isArray(fragment.data)) {
+    return fragment.data.map((issuer: SignatureVerificationFragment) => ({
+      identifier: issuer.did,
+      type,
+    }));
+  }
+  return {
+    identifier: fragment.data?.did,
+    type,
+  };
+};
+
+export const getIdentityProofFragment = (fragments: VerificationFragment[]) => {
```

Comment:
why do we need this function ? why isn't this function called directly within getIdentifier ?

Reply:
no particular reason, wanted to segregate the errors caused by wrong data passed in; instead of handling everything in getIdentifier